vm/src/bytecode.rs

```diff
@@ -252,6 +253,7 @@ impl CodeObject {
         source_path: String,
         first_line_number: usize,
         obj_name: String,
+        is_function_obj: bool,
```

Comment:
Why add this here instead of another flag in `Compiler` like `is_loop`?

Reply:
Both ways would work. I think this has the advantage that we don't have to remember to save and restore the flag when exiting the block. It's also how it's done in the cpython implementation.

Reply:
> Both ways would work. I think this has the advantage that we don't have to remember to save and restore the flag when exiting the block.

True but it has the disadvantage that it makes the information accessible in a wider scope than is currently necessary. So unless there is some feature that depends on distinguish code objects that are functions, I'd prefer just to use a flag for now.

> It's also how it's done in the cpython implementation.

Could you link to the relevant cpython bits?

Reply:
https://github.com/python/cpython/blob/master/Python/compile.c#L2744
https://github.com/python/cpython/blob/master/Python/compile.c#L2758

Reply:
@OddCoincidence I've changed to use boolean flags for checking the invalid return and yield.